### PR TITLE
Unit test file provider

### PR DESCRIPTION
**Pull request by ShaDy03**

### CODE REVIEW

File: IssueTracker/IssueTracker.FileSystem/Repository/BolbData.cs [hunk 1]
```diff
@@ -41,8 +43,11 @@ public async Task UploadFileAsync(Models.File file)
     {
         if (file == null)
             throw new ArgumentException("You don't have files!");
-
-        await ContainerClient.CreateIfNotExistsAsync();
+        if (file.BlobName == String.Empty || file.BlobName == null)
+            throw new ArgumentException("Invalid BlobName");
+        if (file.Content == null)
+            throw new ArgumentException("Invalid Content");
+        //await ContainerClient.CreateIfNotExistsAsync();
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Repository/BolbData.cs]: keep the code clean

File: IssueTracker/IssueTracker.UnitTest/TestInitialization.cs [hunk 2]
```diff
@@ -25,9 +26,32 @@ public static void AssemblyInitialize(TestContext tc)
             BaseClass.TestContext = tc;
 
             IConfigurationFactory cf = new ConfigurationFactory(configuration);
+            var blobConfig = cf.Create<IBolbConfigurationFactory>();
             var metadataconfig = cf.Create<IMetaDataConfiguration>();
+            BaseClass.s_blobData = new BolbData(blobConfig);
             BaseClass.s_metaDataProvider = new MetaData(metadataconfig);
 
+            //begin populate Blob Test
+            var path = "TextForTest.txt";
```
Comment (iiusan) [inline IssueTracker/IssueTracker.UnitTest/TestInitialization.cs]: You need to rethink this.

File: IssueTracker/IssueTracker.UnitTest/TestInitialization.cs [hunk 3]
```diff
@@ -25,9 +26,32 @@ public static void AssemblyInitialize(TestContext tc)
             BaseClass.TestContext = tc;
 
             IConfigurationFactory cf = new ConfigurationFactory(configuration);
+            var blobConfig = cf.Create<IBolbConfigurationFactory>();
             var metadataconfig = cf.Create<IMetaDataConfiguration>();
+            BaseClass.s_blobData = new BolbData(blobConfig);
             BaseClass.s_metaDataProvider = new MetaData(metadataconfig);
 
+            //begin populate Blob Test
```
Comment (iiusan) [inline IssueTracker/IssueTracker.UnitTest/TestInitialization.cs]: Why is this not in a method?

File: IssueTracker/IssueTracker.UnitTest/FileSystemTests/BlobDataTest.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
+﻿using File = IssueTracker.FileSystem.Models.File;
+
+namespace IssueTracker.UnitTest.FileSystemTests;
+
+[TestClass]
+public class BlobDataTest : BaseClass
+{
+    [TestMethod]
+    [Description("Upload Method with Content null, i expected error")]
```
Comment (iiusan) [inline IssueTracker/IssueTracker.UnitTest/FileSystemTests/BlobDataTest.cs:10]: Consistency!!!! Use the same format in all tests.

File: IssueTracker/IssueTracker.UnitTest/BaseClass.cs [hunk 5]
```diff
@@ -8,10 +8,13 @@ public abstract class BaseClass
 {
     public static IIssueRepository IssueData = null!;
     public static TestContext TestContext { get; set; } = null!;
+    internal static string IdFileTest { get; set; } = String.Empty;
+    internal static Stream FileStreamTest { get; set; } = null!;
     public static DataTable? TestDataTable { get; set; }
     public static IParticipantRepository ParticipantData = null!;
```
Comment (iiusan) [inline IssueTracker/IssueTracker.UnitTest/BaseClass.cs]: Consistency!!!! Use the same naming format everywhere.

File: IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs [hunk 6]
```diff
@@ -18,17 +17,18 @@ internal BolbData(IBolbConfigurationFactory config)
     public Task<IEnumerable<Models.File>> GetFilesAsync(IEnumerable<Models.File> files)
     {
         var fileResult = new List<Models.File>();
-
+        if (files.Count() <= 0)
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs]: use files.Any()

File: IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs [hunk 7]
```diff
@@ -18,17 +17,18 @@ internal BolbData(IBolbConfigurationFactory config)
...
+            throw new ArgumentException("I can't get nothing, files is empty!");
         foreach (var file in files)
         {
             var fileAtachment = new Models.File();
             if (file.Id == null || file.Id == string.Empty)
                 throw new ArgumentException("Invalid file id!");
+            if (file.Extension == null || file.Extension == string.Empty)
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs]: use string.IsNullOrEmpty()

File: IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs [hunk 8]
```diff
@@ -41,8 +41,11 @@ public async Task UploadFileAsync(Models.File file)
     {
         if (file == null)
             throw new ArgumentException("You don't have files!");
-
-        await ContainerClient.CreateIfNotExistsAsync();
+        if (file.BlobName == String.Empty || file.BlobName == null)
+            throw new ArgumentException("Invalid BlobName");
+        if (file.Content == null)
+            throw new ArgumentException("Invalid Content");
+        //await ContainerClient.CreateIfNotExistsAsync();
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs]: keep the code clean

File: IssueTracker/IssueTracker.UnitTest/BaseClass.cs [hunk 9]
```diff
@@ -8,10 +9,12 @@ public abstract class BaseClass
 {
     public static IIssueRepository IssueData = null!;
     public static TestContext TestContext { get; set; } = null!;
+    internal static File FileObject { get; set; } = null!;
```
Comment (iiusan) [inline IssueTracker/IssueTracker.UnitTest/BaseClass.cs]: Is the 'Object' suffix required?

File: IssueTracker/IssueTracker.UnitTest/TestInitialization.cs [hunk 10]
```diff
@@ -148,10 +182,10 @@ public static void AssemblyInitialize(TestContext tc)
     [AssemblyCleanup()]
     public static void AssemblyCleanup()
     {
+
```
Comment (iiusan) [inline IssueTracker/IssueTracker.UnitTest/TestInitialization.cs]: Keep the PR clean. Remove this empty line.

File: IssueTracker/IssueTracker.FileSystem/Repository/BolbData.cs (hunk 1 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.FileSystem/Repository/BolbData.cs]: image

File: IssueTracker/IssueTracker.UnitTest/TestInitialization.cs (hunk 3 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.UnitTest/TestInitialization.cs]: image

File: IssueTracker/IssueTracker.UnitTest/TestInitialization.cs (hunk 2 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.UnitTest/TestInitialization.cs]: image

File: IssueTracker/IssueTracker.UnitTest/FileSystemTests/BlobDataTest.cs (hunk 4 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.UnitTest/FileSystemTests/BlobDataTest.cs:10]: image

File: IssueTracker/IssueTracker.UnitTest/BaseClass.cs (hunk 5 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.UnitTest/BaseClass.cs]: These elements were removed. I created an Object of the type File.

File: IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs (hunk 6 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs]: I change this.

File: IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs (hunk 7 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs]: I changed these in both cases.

File: IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs (hunk 8 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.FileSystem/Repository/BolbStorageProvider.cs]: I created a variable where the will be stocked the return.

File: IssueTracker/IssueTracker.UnitTest/BaseClass.cs (hunk 9 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.UnitTest/BaseClass.cs]: I changed this in File.

File: IssueTracker/IssueTracker.UnitTest/TestInitialization.cs (hunk 10 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.UnitTest/TestInitialization.cs]: I removed all the empty lines.

File: IssueTracker/IssueTracker.UnitTest/BaseClass.cs (hunk 5 above)
Comment (iiusan) [reply IssueTracker/IssueTracker.UnitTest/BaseClass.cs]: The comments are always about the last line from the code snippet.

Comment (ShaDy03) [reply IssueTracker/IssueTracker.UnitTest/BaseClass.cs]: i changed in IssueRepository and ParticipantRepository

**Review by iiusan — APPROVED** [review]

**Pull request merged**